### PR TITLE
Update to crossterm-0.25

### PR DESCRIPTION
**Pull request by groves**
I've got a couple PRs that depend on this update. I thought it'd be nice to get the library bump in separately to keep from conflating any issues.

### CODE REVIEW

File: helix-view/src/input.rs [hunk 1]
```diff
@@ -271,6 +273,11 @@ impl From<crossterm::event::Event> for Event {
             crossterm::event::Event::Key(key) => Self::Key(key.into()),
             crossterm::event::Event::Mouse(mouse) => Self::Mouse(mouse.into()),
             crossterm::event::Event::Resize(w, h) => Self::Resize(w, h),
+            crossterm::event::Event::FocusGained => Self::FocusGained,
```
Comment (groves) [inline helix-view/src/input.rs:276]: Focus events are emitted on Windows regardless of any flags being set in crossterm. Don't want to use `unreachable` like in the Paste arm as a result.

File: helix-view/src/input.rs [hunk 2]
```diff
@@ -321,7 +328,14 @@ impl From<crossterm::event::MouseButton> for MouseButton {
 
 #[cfg(feature = "term")]
 impl From<crossterm::event::KeyEvent> for KeyEvent {
-    fn from(crossterm::event::KeyEvent { code, modifiers }: crossterm::event::KeyEvent) -> Self {
+    fn from(
+        crossterm::event::KeyEvent {
+            code,
+            modifiers,
+            kind: _,
+            state: _,
```
Comment (the-mikedavis) [inline helix-view/src/input.rs:336]: these can be discarded a bit more tersely with

```rs
crossterm::event::KeyEvent { code, modifiers, .. }
```

**Comment by archseer** [comment]
Resolved the comment and added a note about the extra state now offered by crossterm

**Comment by sfarq** [comment]
I am using Kitty, does this update mean I can now use some variation of `C-backspace = "delete_word_backward"` and solve https://github.com/helix-editor/helix/issues/1085 ?

**Comment by the-mikedavis** [comment]
I see `C-backspace` being parsed correctly by crossterm with this update on kitty so it should work on kitty at least. I'm not knowledgable about what changed in crossterm in this update so I'm not sure the fix is expected to work on all terminal emulators.

**Comment by groves** [comment]
We might need to enable some of the [keyboard enhancement flags](https://github.com/crossterm-rs/crossterm/blob/10d1dc246dcd708b4902d53a542f732cba32ce99/src/event.rs#L317) to pick up on previously unsupported key combos. I haven't looked into what happens on other terms when emitting those flags, so I didn't want to just enable it here.

**Comment by David-Else** [comment]
@groves Would you be able to add them? It would be much appreciated!

**Pull request merged**